iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2619,6 +2750,50 @@ void IoTHubTransport_MQTT_Common_Unsubscribe_DeviceTwin(IOTHUB_DEVICE_HANDLE han
     }
 }
 
+IOTHUB_CLIENT_RESULT IoTHubTransport_MQTT_Common_GetDeviceTwinAsync(IOTHUB_DEVICE_HANDLE handle, IOTHUB_CLIENT_DEVICE_TWIN_CALLBACK completionCallback, void* callbackContext)
```

Comment:
>IoTHubTransport_MQTT_Common_GetDeviceTwinAsync [](start = 21, length = 46)

This function doesn't look like it's subscribing to Device Twin messages.  So if you haven't set the twin callback then you don't get any twin messages.  Let talk offline, because I may have missed something on this.